### ISSUE TITLE
Details row template with custome javascript

### ISSUE THREAD

**Issue by badarkhan1** [Laravel-Backpack/CRUD#1276]
I am working on an application where i want to show some sensor readings in real time. I have a CRUD for devices and in the details row of each entry for a device i want to show a realtime javascript chart which will show the readings coming from that device, which requires some js to be added along with the template for details row. 

I tried creating a new blade file for the details row and added the js at the end of the body tag but it did'nt work.

I also tried adding adding the js files in master layout file but it didnt work as well, however it did work when i loaded the chart inside dashboard.blade.php, but not working inside details row.

Any help would be appreciated. Thanx

**Comment by tabacitu** [comment]
Hi @badarkhan1 ,

I'm afraid it isn't possible to loads JS in the table cells. They're loaded using an AJAX call, which returns the actual HTML. So even if you _did_ have javascript inside the cell, it wouldn't be run and it would not affect the DOM, it would just be placed inside the HTML by DataTables.

I recommend you [create a custom column type](https://laravel-backpack.readme.io/docs/crud-columns-types#section-roll-your-own) without javascript, but with the indicators you need, something like:
```
<span data-entry-id="{{ $entry->id }}" class="custom-graph">{{ str_limit(strip_tags($entry->{$column['name']}), 80, "[...]") }}</td>
```

and things with Javascript separately. In the example above you'd be able to use the ```$(".custom-graph")``` selector and then the entry-id data attribute. You can use custom JS inside the list view using one of two methods: 
1) by loading a custom blade file with ```$this->crud->setListView('your-view');```
2) by modifying ```public/vendor/backpack/crud/js/list.js```

Hope it helps.
Cheers!

**Comment by censore** [comment]
Hey.
I had to show a khill/lavacharts in my detail row.
And i had to return JS in ajax response.
I ran into a similar problem.
And i made this solution

In MyCrudController.php 

public function showDetailsRow($id)
{
    ...
    return [
        'html'  =>view('vendor.backpack.crud.details_row')->render(),
        'js' => $this->buildDataTable()->render('CalendarChart', 'Calendar', 'callendar-container-'.$id),
}

Then i modified the details_row_logic.blade.php in ajax.done method

 row.child("<div class='table_row_slider_"+data.id+"'>" + data.hasOwnProperty('html')?data.html:data + "</div>", 'no-padding').show();

if(data.hasOwnProperty('lava')){
    $('body').append( $(data.lava) );
}


I hope this will help for someone

> I am working on an application where i want to show some sensor readings in real time. I have a CRUD for devices and in the details row of each entry for a device i want to show a realtime javascript chart which will show the readings coming from that device, which requires some js to be added along with the template for details row.
> 
> I tried creating a new blade file for the details row and added the js at the end of the body tag but it did'nt work.
> 
> I also tried adding adding the js files in master layout file but it didnt work as well, however it did work when i loaded the chart inside dashboard.blade.php, but not working inside details row.
> 
> Any help would be appreciated. Thanx


Hey.
I had to show a khill/lavacharts in my detail row.
And i had to return JS in ajax response.
I ran into a similar problem.
And i made this solution

In MyCrudController.php 

public function showDetailsRow($id)
{
    ...
    return [
        'html'  =>view('vendor.backpack.crud.details_row')->render(),
        'js' => $this->buildDataTable()->render('CalendarChart', 'Calendar', 'callendar-container-'.$id),
}

Then i modified the details_row_logic.blade.php in ajax.done method

 row.child("<div class='table_row_slider_"+data.id+"'>" + data.hasOwnProperty('html')?data.html:data + "</div>", 'no-padding').show();

if(data.hasOwnProperty('lava')){
    $('body').append( $(data.lava) );
}


I hope this will help for someone

**Comment by tabacitu** [comment]
Thanks for sharing your solution @censore ! I’m sure it’ll help someone :-)
